Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -630,263 +852,170 @@ bool UserCmd_Process(uint client, const wstring &wscCmd)
-			string Charname = wstos(charname);
-			string Type = it->second.iDockingModulesInstalled == 0 ? "Docked" : "Carrier";
-			if (Type == "Carrier")
+	
+	const wchar_t* charname = (const wchar_t*)Players.GetActiveCharacterName(iClientID);
+	if (nameToCarrierInfoMap.count(charname)) {
+		idToCarrierInfoMap[iClientID] = &nameToCarrierInfoMap[charname];
+		for (const auto& dockedShipName : idToCarrierInfoMap[iClientID]->dockedShipList) {
+			uint dockedClientID = HkGetClientIdFromCharname(dockedShipName.c_str());
+			if (dockedClientID != -1)
 			{
-				wstring docked = L"";
-				for (map<wstring, wstring>::iterator cit = it->second.mapDockedShips.begin(); cit != it->second.mapDockedShips.end(); cit++)
-				{
-					if (docked != L"")
-						docked += L" | ";
-					docked += cit->first;
-				}
-
-				Type += "[" + to_string(it->second.iDockingModulesInstalled) + "](" + wstos(docked) + ")";
+				PrintUserCmdText(dockedClientID, L"Carrier ship has come online.");
 			}
-			else
-			{
-
-				if (it->second.wscDockedWithCharname.empty())
-				{
-					if (!it->second.mobileDocked)
-						continue;
-					Type += "[" + wstos(L"<Error>") + "]";
-				}
-				else
-					Type += "[" + wstos(it->second.wscDockedWithCharname) + "]";
-			}
-			string State = "";
-
-			if (it->first == 0 || it->first > MAX_CLIENT_ID)
-				State += "Out of range";
-
-			if (find(DOCKcharnames.begin(), DOCKcharnames.end(), charname) != DOCKcharnames.end())
-			{
-				if (State != "")
-					State += " | ";
-				State += "Doubled";
-			}
-
-			if (State == "")
-				State = "Fine";
-
-			if (charname != L"<Error>")
-				DOCKcharnames.push_back(charname);
-
-			Lines.push_back(ID + " " + Charname + " " + Type + " " + State);
 		}
-
-		Lines.push_back("");
-
-
-		vector<string> SERVERlines;
-
-		vector<wstring> SERVERcharnames;
-		vector<uint> IDs;
-		struct PlayerData *pPD = 0;
-		while (pPD = Players.traverse_active(pPD))
+	}
+	else if (nameToDockedInfoMap.count(charname)) {
+		idToDockedInfoMap[iClientID] = &nameToDockedInfoMap[charname];
+		uint carrierClientID = HkGetClientIdFromCharname(idToDockedInfoMap[iClientID]->carrierName.c_str());
+		if (carrierClientID != -1)
 		{
-			uint clientID;
-			try
+			const auto& shipInfo = Archetype::GetShip(Players[iClientID].iShipArchetype);
+			wstring& shipName = HkGetWStringFromIDS(shipInfo->iIdsName);
+			PrintUserCmdText(carrierClientID, L"INFO: Docked %ls, %ls has come online", shipName.c_str(), charname);
+
+			const auto& carrierInfo = Players[carrierClientID];
+			const auto& sysInfo = Universe::get_system(carrierInfo.iSystemID);
+			wstring& sysName = HkGetWStringFromIDS(sysInfo->strid_name);
+			if (carrierInfo.iShipID)
 			{
-				clientID = HkGetClientIdFromPD(pPD);
+				Vector pos;
+				Matrix ori;
+				pub::SpaceObj::GetLocation(carrierInfo.iShipID, pos, ori);
+				wstring& sector = VectorToSectorCoord(sysInfo->id, pos);
+				PrintUserCmdText(iClientID, L"Carrier already in flight, %ls %ls", sysName.c_str(), sector.c_str());

```

Comment:
I am unsure what inflight means and why that is meant to be a problem

Reply:
Yeah, removed the word 'already'. It was just a notification stating "Ship you're docked on is in space, if you undock you'll also be in space next to it"